publish cookies module to allow cookie parsing

lausek opened the pull request:
The parsing of cookie strings is currently not exported for usage in crates. This pull request makes the according functions public.

lausek committed:
publish cookies module to allow cookie parsing

Mahdrentys commented:
Thanks for your contribution! This is a great idea, but I have made some edits to publish the whole `cookies` module instead of just this function, to allow users to use the crate outside a browser.

I've kept `cookies::all_iter` private though, because this use case will already be covered by `cookies::all_raw` (there is an additional allocation with `cookies::all_raw`, but I imagine cookie parsing is not used in performance critical code).

Could you check the `Allow edit from maintainers` on your pull request to allow me to push this changes?

lausek commented:
@Mahdrentys Thanks for the response. I've enabled the "Allow edits by maintainers" flag.

Regarding the exported functions: I don't really see a downside of keeping `all_iter` private, but in the end it is your decision.

Mahdrentys committed:
Publish all functions from `cookies` mod and document them

Don't publish the `cookies::all_iter` function, because this use case is already covered by `cookies::all_raw()`.

Mahdrentys commented:
I apologise for taking so much time for this follow up. I have published the `cookies` module (including the `all_iter` function after all, which is now `all_iter_raw`).

But it doesn't build yet on other targets than `wasm32-unknown-unknown`, because the `js_sys::Date` js api is used to do date processing for the cookie options. I suppose it should use a specific library for that on other targets (but not on wasm to minimise build size). I will work on that.

Mahdrentys commented:
> I suppose it should use a specific library for that on other targets

The issue is that the `Date` object is used in the public API, in the `CookieOptions::expires_at_date` function. I don't really know how to solve this problem, since it woudn't be great to use different types for dates in the api depending on the target.

Boscop commented:
Any update on this? :)

Mahdrentys committed:
Remove `js-sys::Date` from public API

To allow using the crate with non-wasm environments.

Mahdrentys committed:
Revert github actions workflows files

From commit 74efebac8f0c72cfe793eb9687b136832e095646.
They should not be edited in a feature branch.

Mahdrentys commented:
> The issue is that the `Date` object is used in the public API, in the `CookieOptions::expires_at_date` function.

I removed the `js_sys::Date` type from the public API, and added a dependency to `chrono` (only in non-wasm environments).

Mahdrentys commented:
Version 0.2.0 is now on crates.io, sorry for taking so long.
